.github/gke-workflow/nccl/scripts/test.sh

```diff
@@ -0,0 +1,92 @@
+set -x
+
+: "${BENCHMARK:?Must set BENCHMARK}"
+: "${NHOSTS:?Must set NHOSTS}"
+
+ulimit -n 1048576
+
+DATA_MIN="${DATA_MIN:-8G}"
+DATA_MAX="${DATA_MAX:-8G}"
+GPU_PER_NODE="${GPU_PER_NODE:-8}"
+RUN_ITERS="${RUN_ITERS:-20}"
+WARMUP_ITERS="${WARMUP_ITERS:-5}"
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+host_nic="eth0"
+gpu_nics="eth1,eth2,eth3,eth4,eth5,eth6,eth7,eth8"
+find_nics() {
+    local nics_discovered=0
+    local host_nic_discovered=""
+    local gpu_nics_discovered=""
+    for netdev in $(ls /sys/class/net | awk '{print $1}'); do
+        pci_bdf=$(cat "/sys/class/net/${netdev}/device/uevent" 2>/dev/null | grep "PCI_SLOT_NAME" | awk -F '=' '{print $2}' || echo "")
+        if [[ -z "${pci_bdf}" ]]; then
+            continue
+        fi
+        if [[ "${pci_bdf}" =~ 00\:[0-9a-f]{2}\.0 ]]; then
+            host_nic_discovered="${netdev}"
+        elif [[ "" =~ [0|8][6-7d-e]\:00\.0 ]]; then
+            if [[ -z "${gpu_nics_discovered}" ]]; then
+                gpu_nics_discovered="${netdev}"
+            else
+                gpu_nics_discovered="${gpu_nics_discovered},${netdev}"
+            fi
+        fi
+    done
+    if [[ -n "" ]]; then
+        host_nic="${host_nic_discovered}"
+    fi
+    if [[ -n "" ]]; then
+        gpu_nics="${gpu_nics_discovered}"
+    fi
+}
+
+find_nics
```

Comment:
Seems like a no-op (`-n ""`)? Also I am not sure if it is safe to assume that the launcher (where `mpirun` is called) is actually on one of the compute nodes - so introspecting here seems iffy.

Generally I do not think we want to be adopting this kind of logic ourselves, and we should prefer to take GCP's configuration of GCP-specific NCCL parameters.